server/src/datasources/treetracker.datasource.json

```diff
@@ -0,0 +1,10 @@
+{
```

Comment:
We shouldn't upload this

Reply:
And I'm thinking if we need to upload a test datasource file for running test on the Github, maybe we should change the name to another? Maybe there are some risks that we accidentally run the test code locally and do some operation on our real database.

Reply:
Without the treetracker.datasource.json file in the remote repository, GitHub Actions gives me an error when running `npm run build` for the Loopback project. This is the error I got. 

`##[error]src/datasources/treetracker.datasource.ts(3,25): error TS2307: Cannot find module './treetracker.datasource.json'`

There are several options we can look into when running test code, without doing operation on the actual databases. 

- Using an in-memory database. 
- Connect to a local database (not the dev DB). Afterwards, create the PostgreSQL service container in the CI/CD pipeline.

Reply:
@dadiorchen
Can we rework the datasource mechanism to use an env var to determine whether or how to read `treetracker.datasource.json`?
For example, we currently choose between that file and `treetrackerTest.datasource.json`, depending on the value of `process.env.NODE_DB`.

We could either:

- Use an env var to specify the location of the file (defaulting to the current location) and add a version of the file to the CI environment, or
- Not read the file at all if `DB_URL` is set (my preference)

Either way, the file should not be tracked in the git repo.

Reply:
Also, we're importing this file directly in a number of places, which we should avoid.